### PR TITLE
DRILL-8148: Add REST Endpoints to Update OAuth Tokens

### PR DESCRIPTION
**Pull request by cgivre**
# [DRILL-8148](https://issues.apache.org/jira/browse/DRILL-8148): Add REST Endpoints to Update OAuth Tokens

## Description
This PR adds three REST endpoints to Drill to facilitate updating OAuth tokens for REST API calls. 

## Documentation
The following API calls are now available for updating an HTTP Storage Plugin's OAuth tokens if OAuth is used.  These three new endpoints are `POST` requests and accept the tokens as part of the form body. 

* `/storage/{plugin name}/update_access_token`
* `/storage/{plugin name}/update_refresh_token`
* `/storage/{plugin name}/update_oauth_tokens`

These endpoints take as arguments: `access_token`, `refresh_token` or both. 

## Testing
Added unit tests.

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java [hunk 1]
```diff
@@ -200,6 +202,100 @@ public Response enablePlugin(@PathParam("name") String name, @PathParam("val") B
...
+          .entity(message("Unable to add tokens: %s", name))
+          .build();
+      }
+    } catch (PluginException e) {
+      logger.error("Error when adding tokens to {}", name);
+      return Response.status(Status.INTERNAL_SERVER_ERROR)
+        .entity(message("Unable to add tokens: %s", e.getMessage()))
+        .build();
+    }
+  }
```
Comment (jnturton) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java:234]:
```suggestion
  }
  
```

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java [hunk 2]
```diff
@@ -200,6 +202,100 @@ public Response enablePlugin(@PathParam("name") String name, @PathParam("val") B
...
+
   @GET
   @Path("/storage/{name}/update_oath2_authtoken")
```
Comment (jnturton) [inline exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java:300]:
```suggestion
  @Path("/storage/{name}/update_oauth2_authtoken")
```

File: contrib/storage-http/src/test/java/org/apache/drill/exec/store/http/TestOAuthTokenUpdate.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+    HttpApiConfig connectionConfig = HttpApiConfig.builder()
+      .url("http://localhost:" + MOCK_SERVER_PORT + "/getdata")
+      .method("get")
+      .requireTail(false)
+      .inputType("json")
+      .build();
+
+    HttpOAuthConfig oAuthConfig = HttpOAuthConfig.builder()
+      .callbackURL(hostname + "/update_oath2_authtoken")
```
Comment (jnturton) [inline contrib/storage-http/src/test/java/org/apache/drill/exec/store/http/TestOAuthTokenUpdate.java:82]:
```suggestion
      .callbackURL(hostname + "/update_oauth2_authtoken")
```

**Review by jnturton — APPROVED** [review]
Some minor comments but +1

File: exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java (hunk 1 above)
Comment (cgivre) [reply exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java:234]: Fixed

**Comment by cgivre** [comment]
Thanks @jnturton for the review.   I addressed all your comments.  Will merge later.

**Pull request merged**